packages/driver/src/util/tab_activation.ts

```diff
@@ -0,0 +1,43 @@
+import type { ICypress } from '../cypress'
+
+function activateMainTab () {
+  // Don't need to activate the main tab if it already has focus
+  if (document.hasFocus()) return
+
+  return new Promise<void>((resolve) => {
+    const url = `${window.location.origin}${window.location.pathname}`
+
+    // This sends a message on the window that the extension content script
+    // listens for in order to carry out activating the main tab
+    window.postMessage({ message: 'cypress:extension:activate:main:tab', url }, '*')
+
+    function onMessage ({ data, source }) {
+      // only accept messages from ourself
+      if (source !== window) return
+
+      if (data.message === 'main:tab:activated') {
```

Comment:
Also scope this one?
```suggestion
      if (data.message === 'cypress:main:tab:activated') {
```

Reply:
Good idea. Addressed in https://github.com/cypress-io/cypress/pull/28334/commits/639b6d01036d76179115f06434f6678b152f1c62